### PR TITLE
CHANGELOG changes

### PR DESCRIPTION
**Pull request by jasonmp85**
Broke the changes already in release-5.2 into their own section, figure
I'll tag it at the same time (though not actually do any packaging).

### CODE REVIEW

**Comment by jasonmp85** [comment]
I'm not sure whether or how to include #752, #785, #840, #864, #865, #867, and #892. In particular, many of these seem internal, or perhaps part of an incomplete/in-progress feature.

I already edited the included changes to ensure no intra-release bugfixes were included (i.e. stuff that we both introduced and fixed during the course of 6.0 development).

If you want to generate the list of merges yourself, I used `git log --merges v5.2.0...origin/release-6.0 --oneline` to show only merge commits added between the `v5.2.0` tag and the `HEAD` of the `release-6.0` branch (ensure the latter is up to date with a `git fetch`).

If you're feeling really crazy, run `git log --merges v5.2.0...origin/release-6.0 --oneline | egrep -o '#\d+' | tr -d '#' | xargs -L1 -I% open https://github.com/citusdata/citus/pull/%` to open every such pull request in your browser (uses macOS's `open` command). Warning: there are seventy pull requests, so expect a lot of tabs to materialize.

In case you're worried that missed anything, maybe also check out the output of `git log --cherry-pick --first-parent --no-merges origin/release-5.2...origin/release-6.0 --oneline`, which has a better chance of showing anything we cherry-picked.

**Comment by jasonmp85** [comment]
Major needs from reviewer:

  * Did I miss calling out any important features? I feel like more discussion of the worker node changes and colocation might be needed, but it's likely I missed more
  * Of the issues I mention above, which need inclusion/discussion?
  * Is anything minor enough I should just remove it? Perhaps I included something we introduced and fixed during 6.0 development by accident; tell me if so

**Comment by jasonmp85** [comment]
Oh, and of course you can view the full changes since `v5.2.0` [here](https://github.com/citusdata/citus/compare/v5.2.0...release-6.0). Possibly hard to read (and 195 commits), but comprehensive.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,109 @@
...
+
+* Fixes a bug that caused crashes during joins with a `WHERE false` clause
+
+* Fixes a bug triggered by unique violation errors raised in long transactions
+
+* Fixes a bug resulting in multiple registration of transaction callbacks
+
+* Fixes a bug which could result in stale reads of distribution metadata
+
+* TODO: What about #752 ? That's part of an unready feature, right?
```
Comment (sumedhpathak) [inline CHANGELOG.md]: Nope, that's not part of Insert/Select, if thats what you mean. Its basically if you define a function which performs Insert statements. So we should document as bug-fix

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,109 @@
...
+
+* Adds a `create_reference_table` UDF (#892)
+
+### citus v5.2.2 (November 4, 2016) ###
+
+* Adds support for `IF NOT EXISTS` clause of `CREATE INDEX` command
+
+* Adds support for `RETURN QUERY` and `FOR ... IN` PL/pgSQL features
+
+* Extends the router planner to handle more queries
```
Comment (sumedhpathak) [inline CHANGELOG.md]: Which issue was this?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,109 @@
...
+
+* Enhances object name extension to handle long names and large shard counts
+
+* Parallelizes the `master_modify_multiple_shards` UDF
+
+* Changes distributed table creation to error if the target table is not empty
+
+* Changes the `pg_dist_shard.logicalrelid` column from an `oid` to `regclass`
+
+* Adds a true `placementid` column to `pg_dist_shard_placement`
```
Comment (sumedhpathak) [inline CHANGELOG.md]: What does the usage of 'true' represent?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,109 @@
...
+
+* Fixes a bug affecting `COPY` commands using composite types
+
+* Separates worker and master job temporary folders
+
+* Eliminates race condition between distributed modification and repair
+
+* Relaxes the requirement that shard repairs also repair colocated shards
+
+* TODO: What about #785 ?
```
Comment (sumedhpathak) [inline CHANGELOG.md]: We should document. This adds a field to pg_dist_partition called ``colocationid``. This also implements internal functions to determine whether tables are co-located or not, for usage by planner.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,109 @@
...
+
+* TODO: #840
```
Comment (sumedhpathak) [inline CHANGELOG.md]: Worth documenting as making shard move and copy functions co-location aware.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,109 @@
...
+
+* TODO: #865
```
Comment (sumedhpathak) [inline CHANGELOG.md]: Should note addition of pg_dist_local_group table, maybe as 'as preparation for future MX related changes'?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,109 @@
...
+
+* TODO: #864
```
Comment (sumedhpathak) [inline CHANGELOG.md]: Maybe as described "adds the required distributed transaction API for executing commands on workers within a transaction."? Could add note again as preparation for future MX changes.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,109 @@
+### citus v6.0.0 (November 4, 2016) ###
+
+* Adds compatibility with PostgreSQL 9.6, now the recommended version
+
+* Removes the `pg_worker_list.conf` file in favor of a `pg_dist_node` table
```
Comment (sumedhpathak) [inline CHANGELOG.md]: Should we note addition of new node-creation/remove API's?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,109 @@
...
+
+* Removes the `\stage` command and corresponding csql binary in favor of `COPY`
+
+* Removes `copy_to_distributed_table` in favor of first-class `COPY` support
+
+* Adds support for multiple DDL statements within a transaction
+
+* Adds support for certain foreign key constraints
+
+* Adds limited support for `INSERT INTO ... SELECT FROM` commands
```
Comment (sumedhpathak) [inline CHANGELOG.md]: Maybe note something like "parallel and distributed execution of insert/select on colocated tables" ? Just an attempt to highlight the feature a bit more.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,109 @@
...
+
+* TODO: What about #785 ?
+
+* TODO: #840
+
+* TODO: #865
+
+* TODO: #864
+
+* Adds a `create_distributed_table` UDF (#867)
```
Comment (sumedhpathak) [inline CHANGELOG.md]: Should we add something like 'which automatically creates shards, and infers table colocation'? (Not sure if we should mention the colocation here). This function replaces ``master_create_distributed_table/master_create_worker_shards``, so maybe note that as well?

On a related note: Should we add Addition of citus.shard_count configuration value? That is probably part of one of the existing commits, so I don't know if we break them apart.

File: CHANGELOG.md (hunk 8 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Sure, I'll add a line for that.

File: CHANGELOG.md (hunk 9 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Sure, I'll rewrite to be specific about the colocated part.

File: CHANGELOG.md (hunk 3 above)
Comment (jasonmp85) [inline CHANGELOG.md]: As opposed to just using the `oid` of the row, i.e. a real identifier column. I can drop

File: CHANGELOG.md (hunk 4 above)
Comment (jasonmp85) [inline CHANGELOG.md]: I'll add some lines about colocation.

File: CHANGELOG.md (hunk 6 above)
Comment (jasonmp85) [inline CHANGELOG.md]: I'd normally say this one is too internal for mentioning, but as it adds a new metadata table, might as well mention.

File: CHANGELOG.md (hunk 7 above)
Comment (jasonmp85) [inline CHANGELOG.md]: I feel this one is a little too internal to document.

File: CHANGELOG.md (hunk 2 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Join queries with `WHERE false`, I think.

File: CHANGELOG.md (hunk 1 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Are you positive? It appears to be part of the work towards fully supporting parameterized prepared statements, which we're not done with, right?

File: CHANGELOG.md (hunk 2 above)
Comment (jasonmp85) [inline CHANGELOG.md]: That's actually in `v5.2.1`, so I'll remove this line from the `v6.0.0` mentions.

**Comment by jasonmp85** [comment]
I feel like I missed the bugfix for `EXPLAIN EXECUTE`. I'll add that.

**Comment by sumedhpathak** [comment]
I don't have anything specific apart from removing the ToDo's/Questions. After doing that 🚢

**Comment by jasonmp85** [comment]
> After doing that :shipit:

Ack. Cleaning up and merging and tagging everything now.

File: CHANGELOG.md (hunk 8 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 9 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 3 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 5 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 6 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 1 above)
Comment (jasonmp85) [inline CHANGELOG.md]: Done.

**Pull request merged**